core/server/worker/src/main/java/alluxio/worker/grpc/DoraWorkerClientServiceHandler.java

```diff
@@ -101,4 +106,34 @@ public void getStatus(GetStatusPRequest request,
       responseObserver.onError(e);
     }
   }
+
+  @Override
+  public void listStatus(ListStatusPRequest request,
+                         StreamObserver<ListStatusPResponse> responseObserver) {
+    LOG.debug("listStatus is called for {}", request.getPath());
+
+    try {
+      ListOptions options = ListOptions.defaults().setRecursive(
+          request.getOptions().hasRecursive() ? request.getOptions().getRecursive() : false);
+      UfsStatus[] statuses = mWorker.listStatus(request.getPath(), options);
```

Comment:
Do we do chunked / paginated status listing? If the request is recursive and the directories structure is deep, this call can be expensive.

Reply:
That would be an Optimization for the next step.
I have explained this in the commit message:

`This first version of ListStatus from client to Dora worker.
More items are needed:
divide big directory contents into batch limited by max batch size.`

Reply:
I checked again the interfaces of `UnderFileSystem` which is used here.
There is no batching support in `UnderFileSystem`.

Reply:
But, yes, we can do this batching support in the service handler.
I will implement this later.

Reply:
Support from ufs is planned. Haven't got a timeline yet. Metadata sync redesign also needs paginated list and we can probably get help from @tcrain or @elega

Reply:
Thank you, @jja725 .
Let's do this optimization when the UFS is ready.